### ISSUE TITLE
Q about unit fixed effect

### ISSUE THREAD

**Issue by wenddymacro** [KyunghoonBan/rdid#1]
Hi, Kyunghoon, THANKS for your great work. I have a Q on unit FE: in rdid, not need unit FE?

**Comment by KyunghoonBan** [comment]
Hello Wenli,

Thanks for your interest and for reaching out to me.
If you're asking about the need to include unit dummy variables for the typical DID specification, then no, you don't need to manually include unit fixed effects (dummy variables) as long as the treatment dummy and post-treatment dummy are correctly specified for `rdid`.
However, if you're asking about controlling for time-invariant unobserved heterogeneity, typically addressed by adding "fixed effects," we don't have an automated option for that yet.

**Comment by wenddymacro** [comment]
Thanks for your reply. In your simdata, treatment variable = post* unit？ I try to get a TWFE DID result， how to do？
---- Replied Message ----
***@***.***>Date10/12/2024 09:34 ***@***.***> ***@***.***>,
***@***.***>SubjectRe: [KyunghoonBan/rdid] Q about unit fixed effect (Issue #1)
Hello Wenli,
Thanks for your interest and for reaching out to me.
If you're asking about the need to include unit dummy variables for the typical DID specification, then no, you don't need to manually include unit fixed effects (dummy variables) as long as the treatment dummy and post-treatment dummy are correctly specified for rdid.
However, if you're asking about controlling for time-invariant unobserved heterogeneity, typically addressed by adding "fixed effects," we don't have an automated option for that yet.

**Comment by KyunghoonBan** [comment]
Hello Wenli,

First of all, the naïve $2\times 2$ DID results can be obtained by running the following regression:
$`Y_{it} = \alpha_0 + \beta D_{i} + \gamma \mathrm{pos}_{t} + \delta D_{i}  \mathrm{pos}_{t} + \varepsilon_{it},`$
where $\widehat{\delta}$ is the ATT estimate. Note that the variable $D_{i}  \mathrm{pos}_{t}$ is equal to 1 if the observation is both in the post-treatment periods and in the treated group, which is occasionally referred to as the "treatment variable."

To get the TWFE DID result, which is slightly more efficient as we exploit the panel structure, you need to run the following regression:
$`Y_{it} = \alpha_1 + \delta  D_{i}  \mathrm{pos}_{t} + \mu_t + \lambda_i + e_{it},`$
where $\widehat{\delta}$ is the ATT estimate, and $\mu_t$ and $\lambda_i$ are the time and unit dummy variables (or the "fixed-effects"), respectively.

Please feel free to estimate each model by running the following code in Stata:
```
use sim_rdid, clear

gen pos_D = pos*D
reg Y D pos pos_D

gen ind = mod(_n - 1, 100) + 1
gen t_100 = t + 100
reg Y pos*D i.ind i.t_100
```
* Note that the dataset `sim_rdid.dta` currently does not include the individual index, so we need to create `ind` manually for the TWFE exercise.
* Also, note that the estimate $\widehat{\delta}=15.77938$ is included in the estimated RDID bounds, $[9.5525, 19.0281]$.
